inbox/search/backends/imap.py

```diff
@@ -91,17 +91,20 @@ def g():
             encoder = APIEncoder()
 
             with session_scope(self.account_id) as db_session:
-                for imap_uids in self._search(db_session, search_query):
-                    query = (
-                        db_session.query(Message)
-                        .join(ImapUid)
-                        .filter(
-                            ImapUid.account_id == self.account_id,
-                            ImapUid.msg_uid.in_(imap_uids),
+                try:
+                    for imap_uids in self._search(db_session, search_query):
+                        query = (
+                            db_session.query(Message)
+                            .join(ImapUid)
+                            .filter(
+                                ImapUid.account_id == self.account_id,
+                                ImapUid.msg_uid.in_(imap_uids),
+                            )
+                            .order_by(desc(Message.received_date))
                         )
-                        .order_by(desc(Message.received_date))
-                    )
-                    yield encoder.cereal(query.all()) + "\n"
+                        yield encoder.cereal(query.all()) + "\n"
+                except Exception as e:
+                    self.log.error("Error while streaming messages", error=e)
```

Comment:
In previous versions werkzeug would silently swallow exceptions if given a generator to generate a response piece by piece. It now properly fails the whole request in newer versions which is sensible behavior.

Nevertheless we have a test that tests that it does not fail in such case. So I needed to preserve that weird behaviour. At least it would appear now in Rollbar. 

BTW this is inbox search code which we don't use ourselves.